libs/community/langchain_community/text_splitter/semantic_text_splitter.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import Any, List
+
+from langchain_text_splitters import TextSplitter
+
+
+class SemanticTiktokenTextSplitter(TextSplitter):
+    """Splitting text that looks at characters."""
+
+    def __init__(
+        self,
+        *,
+        max_characters: int = 200,
+        model_name: str = "gpt-3.5-turbo",
+        trim_chunks: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        """Create a new TextSplitter."""
+        super().__init__(**kwargs)
+        self.max_characters = max_characters
+        self.trim_chunks = trim_chunks
+        self.model_name = model_name
+
+    def split_text(self, text: str) -> List[str]:
+        """Split incoming text and return chunks."""
+        # First we naively split the large input into a bunch of smaller ones.
+        try:
+            from semantic_text_splitter import TiktokenTextSplitter
+        except ImportError as e:
+            raise ImportError(
+                "Unable to import from semantic-text-splitter."
+                "Please install semantic-text-splitter with "
+                "`pip install -U semantic-text-splitter`."
+            ) from e
+        # Optionally can also have the splitter not trim whitespace for you
+        splitter = TiktokenTextSplitter(self.model_name, trim_chunks=self.trim_chunks)
+        return splitter.chunks(text, self.max_characters)
```

Comment:
Doesn't langchain already support a semantic text splitter based on tiktoken [[source](https://github.com/langchain-ai/langchain/blob/master/libs/text-splitters/langchain_text_splitters/base.py#L220-L265)]?

Reply:
This is based on sementic-text-splitter package referr [https://docs.rs/text-splitter/0.12.1/text_splitter/](https://docs.rs/text-splitter/0.12.1/text_splitter/#method) .

Actually this library also using same method name TiktokenTextSplitter. but the how it is working and giving some better performance 
[TextSplitter Semantic Levels](https://docs.rs/text-splitter/0.12.1/text_splitter/#textsplitter-semantic-levels)

- Characters
- [Unicode Grapheme Cluster Boundaries](https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries)
- [Unicode Word Boundaries](https://www.unicode.org/reports/tr29/#Word_Boundaries)
- [Unicode Sentence Boundaries](https://www.unicode.org/reports/tr29/#Sentence_Boundaries)
- Ascending sequence length of newlines. (Newline is \r\n, \n, or \r) Each unique length of consecutive newline sequences is treated as its own semantic level. So a sequence of 2 newlines is a higher level than a sequence of 1 newline, and so on.